File: Programmers/[12973] 짝지어 제거하기/ONE/12973.java
import java.util.Stack;

class Solution {
    public int solution(String s) {
        int n = s.length();
        Stack<Character> stack = new Stack<>();

        for (int i = 0; i < n; i++) {
            if (!stack.isEmpty() && stack.peek() == s.charAt(i)) {
                stack.pop();
                continue;
            }
            stack.add(s.charAt(i));
        }

        return stack.isEmpty() ? 1 : 0;
    }
}

File: Programmers/[12973] 짝지어 제거하기/KingDonggyu/12973.js
function solution(s) {
  const stack = [];

  for (const ch of s) {
    const top = stack.length - 1;
    if (top > -1 && stack[top] === ch) {
      stack.pop();
      continue;
    }
    stack.push(ch);
  }

  return stack.length ? 0 : 1;
}
